src/main/kotlin/no/skatteetaten/aurora/boober/service/RedeployService.kt

```diff
@@ -24,55 +28,82 @@ class RedeployService(val openShiftClient: OpenShiftClient, val openShiftObjectG
+            return createFailedRedeployResult(it, imageStreamTagResponse, updatedImageStreamResponse)
         }
 
-        if (redeployContext.didImportImage(imageStreamImportResponse)) {
-            return RedeployResult.fromOpenShiftResponses(listOf(imageStreamImportResponse))
+        if (updatedImageStream.isSameImage(imageStream)) {
+            val deploymentRequestResponse = performDeploymentRequest(deploymentSpec)
+            return RedeployResult.fromOpenShiftResponses(listOf(imageStreamTagResponse, updatedImageStreamResponse, deploymentRequestResponse))
         }
 
-        val deploymentRequestResponse = requestDeployment(deploymentSpec)
-        return RedeployResult.fromOpenShiftResponses(listOf(imageStreamImportResponse, deploymentRequestResponse))
+        return RedeployResult.fromOpenShiftResponses(listOf(imageStreamTagResponse, updatedImageStreamResponse))
     }
 
-    private fun createFailedRedeployResult(verificationResult: RedeployContext.VerificationResult, deploymentRequestResponse: OpenShiftResponse) =
-            RedeployResult(success = false, message = verificationResult.message, openShiftResponses = listOf(deploymentRequestResponse))
+    private fun createFailedRedeployResult(message: String?, vararg openShiftResponses: OpenShiftResponse) =
+            RedeployResult(success = false, message = message, openShiftResponses = openShiftResponses.toList())
 
-    private fun importImageStream(deploymentSpec: AuroraDeploymentSpec, redeployContext: RedeployContext): OpenShiftResponse? {
-        val imageStreamImportResource = generateImageStreamImportResource(redeployContext) ?: return null
-        val namespace = deploymentSpec.environment.namespace
-        val command = openShiftClient.createOpenShiftCommand(namespace, imageStreamImportResource)
+    private fun performImageStreamTag(namespace: String, imageStream: ImageStream): OpenShiftResponse? {
+        val imageName = imageStream.findImageName() ?: return null
+        val tagName = imageStream.findTagName() ?: return null
+        val imageStreamTag = ImageStreamTagGenerator().create(imageName, tagName)
+        val command = openShiftClient.createOpenShiftCommand(namespace, imageStreamTag.toJsonNode())
         return try {
             openShiftClient.performOpenShiftCommand(namespace, command)
         } catch (e: OpenShiftException) {
             OpenShiftResponse.fromOpenShiftException(e, command)
         }
     }
 
-    fun generateImageStreamImportResource(redeployContext: RedeployContext): JsonNode? {
-        val imageInformation = redeployContext.findImageInformation()
-        val imageName = redeployContext.findImageName()
-        if (imageInformation != null && imageName != null) {
-            return openShiftObjectGenerator.generateImageStreamImport(imageInformation.imageStreamName, imageName)
+    private fun getUpdatedImageStream(deploymentSpec: AuroraDeploymentSpec): OpenShiftResponse? {
```

Comment:
Hvis jeg forstår hensikten med denne metoden riktig så skal den hente ut den eksisterende ImageStreamen etter ImageStreamTag. Man bør ikke bruke OpenShiftCommand til dette. OpenShiftCommand ble i utgangspunktet laget for å holde de ulike variantene av en ressurs i forbindelse med opprettelse og oppdatering (generert, eksisterende og generert merget med eksisterende = payload). I dette tilfellet bør du gette litt mer direkte, men da gjerne via en ny metode på OpenShiftClient; jeg syntes all aksessering av OpenShift-ressurser fra boober bør gå gjennom denne.

Reply:
Vil det bli noe slikt?

```
fun getImageStream(namespace: String, name: String): ResponseEntity<JsonNode>? {
  val url = "$baseUrl/oapi/v1/namespaces/$namespace/imagestreams/$name"
  return userClient.get(url)
}
```